apps/src/templates/ReCaptchaDialog.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import i18n from '@cdo/locale';
+import Button from './Button';
+import BaseDialog from './BaseDialog';
+import DialogFooter from './teacherDashboard/DialogFooter';
+import Spinner from '../code-studio/pd/components/spinner';
+
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  }
+};
+
+export default class ReCaptchaValidationDialog extends React.Component {
+  static propTypes = {
+    handleSubmit: PropTypes.func.isRequired,
+    handleCancel: PropTypes.func.isRequired,
+    isOpen: PropTypes.bool.isRequired,
+    submitText: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      disableSubmitButton: true,
```

Comment:
Maybe rename this `submitButtonEnabled` to avoid the double negative when setting this to false?